Sources/SourceControl/Repository.swift

```diff
@@ -135,6 +144,9 @@ extension RepositoryProvider {
         fatalError("Unimplemented")
     }
 
+    public func fetch(repository: RepositorySpecifier, to path: AbsolutePath, progress: @escaping GitProgress.Handler) throws {
+        fatalError("Unimplemented")
```

Comment:
This should probably instead call the existing version that doesn't have a progress handler.  Otherwise any existing clients will stop working as soon as this is merged (since they all have `main` dependencies currently).

Reply:
Something like what the corresponding `Repository` extension does below.